src/storage/src/hummock/compactor/compaction_utils.rs

```diff
@@ -366,22 +404,67 @@ pub async fn check_compaction_result(
         Arc::new(TaskProgress::default()),
         compact_io_retry_time,
     );
-    let mut right_iter = UserIterator::new(
+    let right_iter = UserIterator::new(
         SkipWatermarkIterator::from_safe_epoch_watermarks(iter, &compact_task.table_watermarks),
         (Bound::Unbounded, Bound::Unbounded),
         u64::MAX,
         0,
         None,
-        ForwardMergeRangeIterator::default(),
+        del_iter,
+    );
+
+    check_result(left_iter, right_iter).await
+}
+
+pub async fn check_flush_result<I: HummockIterator<Direction = Forward>>(
+    left_iter: UserIterator<I>,
+    existing_table_ids: Vec<StateTableId>,
+    sort_ssts: Vec<SstableInfo>,
+    context: CompactorContext,
+) -> HummockResult<bool> {
+    let mut del_iter = ForwardMergeRangeIterator::default();
+    del_iter.add_concat_iter(sort_ssts.clone(), context.sstable_store.clone());
+    let iter = ConcatSstableIterator::new(
+        existing_table_ids.clone(),
+        sort_ssts.clone(),
+        KeyRange::inf(),
+        context.sstable_store.clone(),
+        Arc::new(TaskProgress::default()),
+        0,
+    );
+    let right_iter = UserIterator::new(
+        iter,
+        (Bound::Unbounded, Bound::Unbounded),
+        u64::MAX,
+        0,
+        None,
+        del_iter,
     );
+    check_result(left_iter, right_iter).await
+}
+
+async fn check_result<
+    I1: HummockIterator<Direction = Forward>,
+    I2: HummockIterator<Direction = Forward>,
+>(
+    mut left_iter: UserIterator<I1>,
+    mut right_iter: UserIterator<I2>,
+) -> HummockResult<bool> {
     left_iter.rewind().await?;
     right_iter.rewind().await?;
     while left_iter.is_valid() && right_iter.is_valid() {
-        assert_eq!(left_iter.key(), right_iter.key());
-        assert_eq!(left_iter.value(), right_iter.value());
+        if left_iter.key() != right_iter.key() {
```

Comment:
Print the key, value, or count that differs?

Reply:
It is difficult to print both count and the content of compact task

Reply:
I mean, when key differs, print keys in the assert.

Reply:
done